src/exactio/exactio_exanic.c

```diff
@@ -35,14 +35,22 @@ typedef enum {
 } exactio_exa_mod_t;
 
 
+static int set_exanic_params(exanic_t *exanic, char* device, int port_number,
+                             bool promisc, bool kernel_bypass);
 
-static void exa_destroy(eio_stream_t* this)
+void exa_destroy(eio_stream_t* this)
 {
     exa_priv_t* priv = IOSTREAM_GET_PRIVATE(this);
+
     if(priv->closed){
         return;
     }
 
+    if(set_exanic_params(priv->rx_nic, priv->rx_dev, priv->rx_port, 0, 0))
+    {
+        printf ("Unable to restore promisc and kernel bypass mode\n");
```

Comment:
Please use the ch_log_xxx functions for error reporting. 
ch_log_error() / ch_log_warning() / ch_log_fatal() are all options depending on the severity.

Reply:
I used the `ch_log_warn` method.  What about the other fprintf in the file ?